crates/infra/utils/src/errors/mod.rs

```diff
@@ -4,14 +4,15 @@ use crate::paths::PathExtensions;
 use anyhow::{bail, Result};
 use ariadne::{Color, Label, Report, ReportKind, Source};
 
-#[derive(Debug)]
+#[allow(clippy::len_without_is_empty)]
```

Comment:
I wonder why allowing `len_without_is_empty` specifically here?
If it is not worth it, should we keep the rule disabled?

Personally, I don't mind enabling the rule, if it results in more standard/idiomatic code.

Reply:
In general, if we have a data structure that exposes a `pub fn len` it also makes sense to expose an `is_empty` for the clearer alternative for the `== 0` check (and in general Rust does it, see https://std.rs/is_empty).

However, in this case we use the `fn into_result` as the combinator/"monadically" that does this internally already. 

This is a minor thing here, so I decided to silence it in this case to not force any decisions but if you're fine with it, I have pushed 1be5b8dfe5d539caf92e264faf26dff0917d68fb with a simple `pub fn is_empty` impl.